app/services/one_off/fhir/prescription_drug_exporter.rb

```diff
@@ -0,0 +1,118 @@
+require "fhir_models"
+
+module OneOff
+  module Fhir
+    class PrescriptionDrugExporter
+      attr_reader :prescription_drug
+
+      def initialize(prescription_drug)
+        @prescription_drug = prescription_drug
+      end
+
+      def export
+        FHIR::MedicationRequest.new(
+          contained: [
+            FHIR::Medication.new(
+              identifier: FHIR::Identifier.new(
+                value: prescription_drug.id
+              ),
+              code: FHIR::CodeableConcept.new(
+                coding: FHIR::Coding.new(
+                  system: "http://www.nlm.nih.gov/research/umls/rxnorm",
+                  code: prescription_drug.rxnorm_code,
+                  display: prescription_drug.name
+                )
+              )
+            )
+          ],
+          status: prescription_drug.is_deleted ? "ended" : "active",
+          intent: "proposal",
+          medicationReference: FHIR::Reference.new(
+            id: FHIR::Medication.new(
+              identifier: FHIR::Identifier.new(
+                value: prescription_drug.id
+              ),
+              code: FHIR::CodeableConcept.new(
+                coding: FHIR::Coding.new(
+                  system: "http://www.nlm.nih.gov/research/umls/rxnorm",
+                  code: prescription_drug.rxnorm_code,
+                  display: prescription_drug.name
+                )
+              )
+            )
+          ),
+          dosageInstruction: dosage_instruction,
+          dispenseRequest: dispense_request,
+          performer: FHIR::Reference.new(
+            id: FHIR::Organization.new(
+              identifier: FHIR::Identifier.new(
+                value: prescription_drug.facility_id
+              )
+            )
+          ),
+          subject: FHIR::Reference.new(
+            id: FHIR::Patient.new(
+              identifier: FHIR::Identifier.new(
+                value: prescription_drug.patient_id
+              )
+            )
+          ),
+          meta: FHIR::Meta.new(
+            lastUpdated: prescription_drug.device_updated_at.iso8601,
+            createdAt: prescription_drug.device_created_at.iso8601
+          )
+        )
+      end
+
+      def dosage_instruction
+        if prescription_drug.frequency.nil?
+          return nil
+        end
+        [
+          FHIR::Dosage.new(
+            doseAndRate: [
+              FHIR::Dosage::DoseAndRate.new(
+                doseQuantity: FHIR::Quantity.new(
+                  value: prescription_drug.dosage.split("MG").first,
```

Comment:
[nonblocking, style]
```suggestion
                  value: prescription_drug.dosage.delete_suffix("MG"),
```

Reply:
I realised this suggestion wouldn't universally work. But the `split` won't work universally either.

Here are all the variations of values I have found looking at metabase.

There are over 100 occurrences of each of these values in Bangladesh and India:
```
500 mg TDS
12.5 mg
30mg5
0+0+1mg
50+mg
.5mg
mg

mg20
2000 mg
6.5mg
```

And these are not even getting into corner cases with a few occurrences like:
```
24iu
50plus
20+0+16unit
5tab
,3mg
2omg
```

Notably `50MGOD` is a format that only exists in sandbox data.

Newer data seems to be a lot more consistent since the app doesn't make dosage input very freeform anymore, but it's still one of `50 mg` or `50 mg OD`. But I wonder if we should be trying to parse this at all. One compromise I can think of is conditionally parsing dosage only if it matches the aforementioned format.

Reply:
:/ Yeah. This also was a hack to get things working quickly.
Conditionally pasring sounds fine. Will log a warning

Reply:
Also, just came across `2 puff`, `BD` on metabse

Reply:
This is done